### PR TITLE
Web3Errors Refactoring

### PR DESCRIPTION
**Pull request by jdevcs**
## Description

Fixes https://github.com/ChainSafe/web3.js/issues/4988 and https://github.com/ChainSafe/web3.js/issues/4989

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run dtslint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:cov` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` with success.
- [ ] I have tested the built `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.

### CODE REVIEW

File: packages/web3-core/package.json [hunk 1]
```diff
@@ -25,6 +25,7 @@
 		"test:integration": "jest --config=./test/integration/jest.config.js --passWithNoTests"
 	},
 	"dependencies": {
+		"web3-errors": "1.0.0-alpha.0",
```
Comment (avkos) [inline packages/web3-core/package.json:28]: looks like should be 4x version

File: packages/web3-errors/package.json [hunk 2]
```diff
@@ -0,0 +1,41 @@
+{
+	"name": "web3-errors",
+	"version": "1.0.0-alpha.0",
```
Comment (avkos) [inline packages/web3-errors/package.json:3]: also should be 4x. Isn't it?

File: packages/web3-eth-accounts/package.json [hunk 3]
```diff
@@ -40,6 +40,7 @@
 		"jest-when": "^3.4.1"
 	},
 	"dependencies": {
+		"web3-errors": "1.0.0-alpha.0",
```
Comment (avkos) [inline packages/web3-eth-accounts/package.json:43]: should be 4.x version

File: packages/web3-eth-contract/package.json [hunk 4]
```diff
@@ -25,6 +25,7 @@
 		"test:integration": "jest --config=./test/integration/jest.config.js --runInBand --forceExit"
 	},
 	"dependencies": {
+		"web3-errors": "1.0.0-alpha.0",
```
Comment (avkos) [inline packages/web3-eth-contract/package.json:28]: should be 4.x version

File: packages/web3-eth-ens/package.json [hunk 5]
```diff
@@ -41,6 +41,7 @@
 		"typescript": "^4.5.2"
 	},
 	"dependencies": {
+		"web3-errors": "1.0.0-alpha.0",
```
Comment (avkos) [inline packages/web3-eth-ens/package.json:44]: should be 4.x version

File: packages/web3-eth-iban/package.json [hunk 6]
```diff
@@ -39,6 +39,7 @@
 		"typescript": "^4.5.2"
 	},
 	"dependencies": {
+		"web3-errors": "1.0.0-alpha.0",
```
Comment (avkos) [inline packages/web3-eth-iban/package.json:42]: should be 4.x version

File: packages/web3-eth/package.json [hunk 7]
```diff
@@ -40,6 +40,7 @@
 		"web3-providers-http": "4.0.0-alpha.1"
 	},
 	"dependencies": {
+		"web3-errors": "1.0.0-alpha.0",
```
Comment (avkos) [inline packages/web3-eth/package.json:43]: should be 4.x version

File: packages/web3-providers-http/package.json [hunk 8]
```diff
@@ -41,6 +41,7 @@
 		"typescript": "^4.5.2"
 	},
 	"dependencies": {
+		"web3-errors": "1.0.0-alpha.0",
```
Comment (avkos) [inline packages/web3-providers-http/package.json:44]: should be 4.x version

File: packages/web3-providers-ipc/package.json [hunk 9]
```diff
@@ -39,6 +39,7 @@
 		"typescript": "^4.5.2"
 	},
 	"dependencies": {
+		"web3-errors": "1.0.0-alpha.0",
```
Comment (avkos) [inline packages/web3-providers-ipc/package.json:42]: should be 4.x version

File: packages/web3-providers-ws/package.json [hunk 10]
```diff
@@ -41,6 +41,7 @@
 		"typescript": "^4.5.2"
 	},
 	"dependencies": {
+		"web3-errors": "1.0.0-alpha.0",
```
Comment (avkos) [inline packages/web3-providers-ws/package.json:44]: should be 4.x version

File: packages/web3-utils/package.json [hunk 11]
```diff
@@ -40,6 +40,7 @@
 		"typescript": "^4.5.2"
 	},
 	"dependencies": {
+		"web3-errors": "1.0.0-alpha.0",
```
Comment (avkos) [inline packages/web3-utils/package.json:43]: should be 4.x version

File: packages/web3/package.json [hunk 12]
```diff
@@ -56,6 +56,7 @@
 		"web3-eth-ens": "4.0.0-alpha.0",
 		"web3-eth-iban": "4.0.0-alpha.0",
 		"web3-net": "^4.0.0-alpha.0",
-		"web3-utils": "4.0.0-alpha.1"
+		"web3-utils": "4.0.0-alpha.1",
+		"web3-errors": "1.0.0-alpha.0"
```
Comment (avkos) [inline packages/web3/package.json:60]: should be 4.x version

File: packages/web3-common/package.json [hunk 13]
```diff
@@ -27,7 +27,8 @@
 	"dependencies": {
 		"web3-utils": "4.0.0-alpha.1",
 		"web3-validator": "^0.1.0-alpha.0",
-		"web3-eth-iban": "4.0.0-alpha.0"
+		"web3-eth-iban": "4.0.0-alpha.0",
+		"web3-errors": "1.0.0-alpha.0"
```
Comment (avkos) [inline packages/web3-common/package.json:31]: should be 4.x version

**Review by avkos — CHANGES_REQUESTED** [review]

**Review by luu-alex — APPROVED** [review]

File: packages/web3-errors/package.json (hunk 2 above)
Comment (luu-alex) [reply packages/web3-errors/package.json:3]: When we create new packages even for 4.x, we start at 1.0.0 for [example](https://github.com/ChainSafe/web3.js/pull/4991/files/c57bb102daee6e7f0c619df0a7ab6b3433fccc84#diff-a6b43d54939f69b811f969bd81d65714cfbb7acf00b4d419d65d13b5e62f9a1bL29) the common package

**Review by nikoulai — APPROVED** [review]

File: packages/web3-core/package.json (hunk 1 above)
Comment (jdevcs) [reply packages/web3-core/package.json:28]: There wasn't any errors package released in past so it will start from 1.0.0

**Review by nazarhussain — APPROVED** [review]

**Review by avkos — APPROVED** [review]

**Pull request merged**